### PR TITLE
80 system admin restore deleted records

### PR DESCRIPTION
**Pull request by rik86future**

### CODE REVIEW

File: app/controllers/needs_controller.rb [hunk 1]
```diff
@@ -89,6 +110,31 @@ def request_destroy_note
...
   end
 
+  def restore_need
+    need = policy_scope(Need).deleted.where(id: params[:id])
+    if need.exists?
+      need_name = need.first.category
+      Rails.logger.info("Restored need '#{params[:id]}'")
+      need.first.restore(:recursive => true)
+      redirect_to deleted_needs_path(order: "deleted_at", order_dir: "DESC"), 
+        notice: "Restored '#{need_name}' see <a href='/needs/#{need.first.id}'>here</a>"
```
Comment (jamiebuckley) [inline app/controllers/needs_controller.rb:120]: could use the URL helpers here, e.g. `#{need_path(need.first.id)}`

Comment (jamiebuckley) [reply app/controllers/needs_controller.rb:120]: And potentially:
`view_context.link_to` helper as well

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
       <!-- Notifications -->
       <% flash.each do |type, msg| %>
-        <div class="<%= type %>"><%= msg %></div>
+        <div class="<%= type %>"><%= msg.html_safe %></div>
```
Comment (jamiebuckley) [inline app/views/layouts/application.html.erb:36]: Perhaps I'm being paranoid, but wonder if we should run this through something like https://github.com/rgrove/sanitize first

File: app/controllers/needs_controller.rb (hunk 1 above)
Comment (rik86future) [reply app/controllers/needs_controller.rb]: done

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (rik86future) [reply app/views/layouts/application.html.erb]: fair, I used the built in SanitizeHelper instead

**Comment by rik86future** [comment]
[System admin restore "deleted" records](https://trello.com/c/3M3eMEnC/80-system-admin-restore-deleted-records)

**Pull request merged**